test/checks/liquid_tag_test.rb

```diff
@@ -51,4 +51,472 @@ def test_ignores_inside_liquid_tag
+      "templates/index.liquid" => <<~END,
+        {% if collection.image.size != 0 %}
+          <div>hello</div>
+        {% elsif collection.products.first.size != 0 and collection.products.first.media != empty %}
+          <div>goodbye</div>
+        {% else %}
+          <div>hello</div>
+        {% endif %}
+      END
+    }
+
+    sources = fix_theme(
+      ThemeCheck::LiquidTag.new(min_consecutive_statements: 4),
+      "templates/index.liquid" => <<~END,
+        {% if collection.image.size != 0 %}
+          <div>hello</div>
+        {% elsif collection.products.first.size != 0 and collection.products.first.media != empty %}
+          <div>goodbye</div>
+        {% else %}
+          <div>hello</div>
+        {% endif %}
+      END
+    )
+
+    sources.each do |path, source|
+      assert_equal(expected_sources[path], source)
+    end
+  end
+
+  def test_doesnt_correct_if_with_variables
+    expected_sources = {
+      "templates/index.liquid" => <<~END,
+        {% if collection.image.size != 0 %}
+          Hello {{ name }}
+        {% elsif collection.products.first.size != 0 and collection.products.first.media != empty %}
+          Goodbye {{ name }}
+        {% else %}
+          Welcome {{ name }}
+        {% endif %}
+      END
+    }
+
+    sources = fix_theme(
+      ThemeCheck::LiquidTag.new(min_consecutive_statements: 4),
+      "templates/index.liquid" => <<~END,
+        {% if collection.image.size != 0 %}
+          Hello {{ name }}
+        {% elsif collection.products.first.size != 0 and collection.products.first.media != empty %}
+          Goodbye {{ name }}
+        {% else %}
+          Welcome {{ name }}
+        {% endif %}
+      END
+    )
+
+    sources.each do |path, source|
+      assert_equal(expected_sources[path], source)
+    end
+  end
+
+  def test_new_line_inside_tag
+    expected_sources = {
+      "templates/index.liquid" => <<~END,
+        {% liquid
+          assign a = 1
+          assign b = 2
+          assign c = 3
+          assign x = 1
+        %}
+      END
+    }
+
+    sources = fix_theme(
+      ThemeCheck::LiquidTag.new(min_consecutive_statements: 4),
+      "templates/index.liquid" => <<~END,
+        {% assign a = 1
+        %}
+        {%
+          assign b = 2
+        %}
+        {%
+          assign c = 3 %}
+        {% assign x = 1 %}
+      END
+    )
+
+    sources.each do |path, source|
+      assert_equal(expected_sources[path], source)
+    end
+  end
+
+  def test_spacing_in_tags
+    expected_sources = {
+      "templates/index.liquid" => <<~END,
+        {% liquid
+          assign a = 1
+          assign b = 2
+          assign c = 3
+          assign x = 1
+        %}
+      END
+    }
+
+    sources = fix_theme(
+      ThemeCheck::LiquidTag.new(min_consecutive_statements: 4),
+      "templates/index.liquid" => <<~END,
+        {% assign a = 1     %}
+        {%      assign b = 2%}
+        {%assign c = 3%}
+        {%   assign x = 1   %}
```

Comment:
Not sure, but maybe we need tests cases for `{% case %}{% when %}` as well? It might show up as a parent node with child nodes in the AST. Similar to the if statement.